scripts/src/java/org/oppia/android/scripts/docs/KDocValidityCheck.kt

```diff
@@ -158,33 +182,52 @@ private fun File.asString(): String {
  * @param file the file to be checked for the KDoc presence
  * @param kDocNotRequiredAnnotationEntryList the list of annotation entries which when present
  *     on an element, the element does not needs to be checked for a KDoc.
- * @return whether the element is missing a KDoc
+ * @return Returns list of elements missing KDoc
  */
-private fun elementIterator(
+private fun recursiveKdocPresenceChecker(
   elem: KtElement,
   file: File,
-  kDocNotRequiredAnnotaionEntryList: List<String>
-): Boolean {
-  if (elem is KtClass) {
-    val classKdocMissing = checkIfKDocMissing(elem, file, kDocNotRequiredAnnotaionEntryList)
-    val memberMissingKdoc = elem.declarations.fold(initial = false) { isMissingKdoc, childElem ->
-      val childMissingKdoc = elementIterator(childElem, file, kDocNotRequiredAnnotaionEntryList)
-      isMissingKdoc || childMissingKdoc
+  kDocNotRequiredAnnotationEntryList: List<String>
+): List<Pair<File, Int?>?> {
+  when {
+    elem is KtObjectDeclaration && elem.isCompanion() -> {
+      val memberMissingKdoc = elem.declarations.flatMap { childElem ->
+        recursiveKdocPresenceChecker(
+          childElem,
+          file,
+          kDocNotRequiredAnnotationEntryList
+        )
+      }
+      return memberMissingKdoc
+    }
+    elem is KtClass -> {
+      val classKdocMissing = checkIfKDocIsMissing(elem, file, kDocNotRequiredAnnotationEntryList)
+      val memberMissingKdoc = elem.declarations.flatMap { childElem ->
+        recursiveKdocPresenceChecker(
+          childElem,
+          file,
+          kDocNotRequiredAnnotationEntryList
+        )
+      }
+      return listOf(listOf(classKdocMissing), memberMissingKdoc).flatten()
```

Comment:
Can be simplified I think to:

```suggestion
      return memberMissingKdoc + classKdocMissing
```

Ditto elsewhere.

Reply:
Done.